modules/identity/src/Volo.Abp.Identity.Application/Volo/Abp/Identity/ProfileAppService.cs

```diff
@@ -20,9 +21,13 @@ public ProfileAppService(IdentityUserManager userManager)
 
         public virtual async Task<ProfileDto> GetAsync()
         {
-            return ObjectMapper.Map<IdentityUser, ProfileDto>(
-                await UserManager.GetByIdAsync(CurrentUser.GetId())
-            );
+            var currentUser = await UserManager.GetByIdAsync(CurrentUser.GetId());
+
+            var profile = ObjectMapper.Map<IdentityUser, ProfileDto>(currentUser);
+            profile.IsExternalLoggedIn = currentUser.IsExternal;
+            profile.HasPassword = currentUser.PasswordHash != null;
+
+            return profile;
```

Comment:
After changing the mapping to fully use automapper, you can rollback changes in this method.